### ISSUE TITLE
TypeError: cannot unpack non-iterable NoneType object when protobuf module does not exist

### ISSUE THREAD

**Issue by jckegelman** [nelfin/pylint-protobuf#6]
Hello, I found that pylint-protobuf raises a `TypeError` when the protobuf module does not exist.

It can be reproduced by running `pylint` with the `pylint-protobuf` plugin on the following code:
```python
import not_a_module_pb2
```

This raises the following exception:
```
Traceback (most recent call last):
  File ".../pylint/__main__.py", line 7, in <module>
    pylint.run_pylint()
  File ".../pylint/__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File ".../pylint/lint.py", line 1628, in __init__
    linter.check(args)
  File ".../pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File ".../pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File ".../pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File ".../pylint/utils.py", line 1303, in walk
    self.walk(child)
  File ".../pylint/utils.py", line 1300, in walk
    cb(astroid)
  File ".../pylint_protobuf/__init__.py", line 413, in visit_import
    self._import_node(node, modname, alias)
  File ".../pylint_protobuf/__init__.py", line 421, in _import_node
    new_scope, new_fields = import_(node, modname, self._scope, self._type_fields)
  File ".../pylint_protobuf/__init__.py", line 340, in import_
    new_scope, new_fields = _do_import(node, modname, scope, type_fields)
TypeError: cannot unpack non-iterable NoneType object
```

It looks like the problem arises from the following lines in the `_do_import` method:
https://github.com/nelfin/pylint-protobuf/blob/ee6ced1d635a2c81c1adb3987c8dac33f4d45e4d/pylint_protobuf/__init__.py#L296-L301

the `import_` method is expecting `_do_import` to return a tuple:
https://github.com/nelfin/pylint-protobuf/blob/ee6ced1d635a2c81c1adb3987c8dac33f4d45e4d/pylint_protobuf/__init__.py#L343

I'm not sure what to do here instead. I briefly tried returning a tuple of `(None, None)`, but that causes more issues later on. Any ideas?

**Comment by nelfin** [comment]
I think it's fine just to pass through the old scope and field mappings unchanged: if we were unable to import the module, then we are unable to update the state. Those lines have survived a couple of refactors and I obviously haven't gotten around to exercising them recently.

**Comment by nelfin** [comment]
Let me know what you think with git master or `pip install pylint-protobuf==0.4.dev2`. If you're happy with it as well I'll tag this up as 0.4 and release these changes to PyPI.

**Comment by jckegelman** [comment]
Thanks! The patch does indeed fix the original test case.

I've now run into a new, related issue with the latest revision when importing a missing module as an "alias":

Running `pylint` on
```python
import missing_module_pb2 as my_module
```

raises the following exception:
```
Traceback (most recent call last):
  File ".../pylint/__main__.py", line 7, in <module>
    pylint.run_pylint()
  File ".../pylint/__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File ".../pylint/lint.py", line 1628, in __init__
    linter.check(args)
  File ".../pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File ".../pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File ".../pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File ".../pylint/utils.py", line 1303, in walk
    self.walk(child)
  File ".../pylint/utils.py", line 1300, in walk
    cb(astroid)
  File ".../pylint_protobuf/__init__.py", line 371, in visit_import
    self._import_node(node, modname, alias)
  File ".../pylint_protobuf/__init__.py", line 383, in _import_node
    diff = new_scope[modname]
KeyError: 'missing_module_pb2'
```

I can open a new issue if you'd like.

Thanks for the quick responses!

**Comment by nelfin** [comment]
No need to open a new issue, it's arguably the same "importing a missing module should not affect the ability to use the linter".

**Comment by jckegelman** [comment]
I checked the latest master and it fixes this issue - thanks!

**Comment by nelfin** [comment]
Great, thanks!

**Comment by nelfin** [comment]
Whoops, 6bb6fa2 was supposed to reference #7, not #6. Oh well.
